src/aleph/sdk/security.py

```diff
@@ -0,0 +1,46 @@
+from importlib import import_module
+from typing import Any, Union
+
+from aleph_message.models import AlephMessage
+
+from aleph.sdk.chains.common import get_verification_buffer
+from aleph.sdk.query.responses import Post
+
+validator_chains_map = {
+    # TODO: Add AVAX
+    "ETH": "ethereum",
+    "SOL": "sol",
+    "CSDK": "cosmos",
+    "DOT": "substrate",
+    "NULS2": "nuls2",
+    "TEZOS": "tezos",
+}
+
+
+def try_import_verify_signature(chain: str) -> Any:
+    """Try to import a chain signature validator."""
+    try:
+        return import_module(f"aleph.sdk.chains.{chain}").verify_signature
+    except (ImportError, AttributeError):
+        return None
+
+
+validators = {
+    key: try_import_verify_signature(value)
+    for key, value in validator_chains_map.items()
+}
+
+
+def verify_message_signature(message: Union[AlephMessage, Post]) -> None:
+    """Verify the signature of a message."""
```

Comment:
Can you be more detailed ? Repeating the name of the function is not very useful...

```suggestion
    """Verify the signature of a message, raise an error if invalid or unsupported.
    A BadSignatureError is raised when the signature is incorrect.
    A ValueError is raised when the chain is not supported or required dependencies are  missing.
    """
```